### PR TITLE
Make getPropertyAsInt throw

### PR DESCRIPTION
**Pull request by externl**
This PR updates `getPropertyAsInt` to throw a `PropertyException` in the event that the property cannot be converted to an integer.

### CODE REVIEW

File: cpp/include/Ice/Properties.h [hunk 1]
```diff
@@ -82,15 +82,17 @@ namespace Ice
         std::string getPropertyWithDefault(std::string_view key, std::string_view value) noexcept;
 
         /**
-         * Get a property as an integer. If the property is not set, 0 is returned.
+         * Get a property as an integer. If the property is not set, 0 is returned. Throws PropertyException if
```
Comment (bernardnormier) [inline cpp/include/Ice/Properties.h:85]: What the function throws belongs to the @throws tag, not the summary.

File: cpp/include/Ice/Properties.h [hunk 2]
```diff
@@ -82,15 +82,17 @@ namespace Ice
...
          * @param key The property key.
          * @return The property value interpreted as an integer.
          * @see #setProperty
          */
-        int getPropertyAsInt(std::string_view key) noexcept;
+        int getPropertyAsInt(std::string_view key);
 
         /**
-         * Get an Ice property as an integer. If the property is not set, its default value is returned.
+         * Get an Ice property as an integer. If the property is not set, its default value is returned. Throws
```
Comment (bernardnormier) [inline cpp/include/Ice/Properties.h:94]: Ditto

File: cpp/include/Ice/Properties.h [hunk 3]
```diff
@@ -99,13 +101,14 @@ namespace Ice
         int getIcePropertyAsInt(std::string_view key);
 
         /**
-         * Get a property as an integer. If the property is not set, the given default value is returned.
+         * Get a property as an integer. If the property is not set, the given default value is returned. Throws
+         * PropertyException if the property value is not a valid integer.
```
Comment (bernardnormier) [inline cpp/include/Ice/Properties.h:105]: Ditto

File: csharp/src/Ice/Properties.cs [hunk 4]
```diff
@@ -165,15 +165,17 @@ public string getPropertyWithDefault(string key, string value)
 
     /// <summary>
     /// Get a property as an integer.
-    /// If the property is not set, 0 is returned.
+    /// If the property is not set, 0 is returned. Throws PropertyException if the property
```
Comment (bernardnormier) [inline csharp/src/Ice/Properties.cs:168]: The summary should not describe what the method throws.

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Properties.java [hunk 5]
```diff
@@ -215,6 +216,7 @@ public int getPropertyAsInt(String key) {
      *
      * @param key The property key.
      * @return The property value interpreted as an integer, or the default value.
+     * @throws PropertyException Raised if the property value is not a valid integer.
```
Comment (bernardnormier) [inline java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Properties.java:219]: Would be good to document what we throw if the key doesn't designate a valid Ice property.

File: php/test/Ice/properties/Client.php [hunk 6]
```diff
@@ -70,5 +70,15 @@ function run($args)
             test($ex->getMessage() == "unknown Ice property: Ice.UnknownProperty");
         }
         echo "ok\n";
+
+        echo "testing that trying to read a non-numeric value as an int throws... ";
+        try {
```
Comment (bernardnormier) [inline php/test/Ice/properties/Client.php:75]: The try block should not include the "arrange" part of the test.

File: ruby/test/Ice/properties/Client.rb [hunk 7]
```diff
@@ -101,5 +101,15 @@ def run(args)
         end
         puts "ok"
 
+        print "testing that trying to read a non-numeric value as an int throws... "
+        begin
+            properties = Ice.createProperties(args)
```
Comment (bernardnormier) [inline ruby/test/Ice/properties/Client.rb:106]: Same here.

File: swift/src/Ice/Properties.swift [hunk 8]
```diff
@@ -34,14 +34,14 @@ public protocol Properties: AnyObject {
     /// - parameter _: `String` The property key.
     ///
     /// - returns: `Int32` - The property value interpreted as an integer.
```
Comment (bernardnormier) [inline swift/src/Ice/Properties.swift:36]: Need to document exception.

File: swift/src/Ice/Properties.swift [hunk 9]
```diff
@@ -34,14 +34,14 @@ public protocol Properties: AnyObject {
...
-    func getPropertyAsInt(_ key: String) -> Int32
+    func getPropertyAsInt(_ key: String) throws -> Int32
 
     /// Get an Ice property as an integer. If the property is not set,its default value is returned.
     ///
     /// - parameter key: `String` The property key.
     ///
     /// - returns: `Int32` - The property value interpreted as an integer, or the default value.
-    func getIcePropertyAsInt(_ key: String) -> Int32
+    func getIcePropertyAsInt(_ key: String) throws -> Int32
```
Comment (bernardnormier) [inline swift/src/Ice/Properties.swift:44]: Ditto.

**Review by bernardnormier — APPROVED** [review]

**Review by pepone — APPROVED** [review]
Seems you still have to update JavaScript

File: csharp/src/Ice/Properties.cs (hunk 4 above)
Comment (pepone) [reply csharp/src/Ice/Properties.cs:168]: Still have to move the exception docs to `exception` tag

File: csharp/src/Ice/Properties.cs [hunk 10]
```diff
@@ -165,15 +165,17 @@ public string getPropertyWithDefault(string key, string value)
...
     /// </summary>
     /// <param name="key">The property key.</param>
     /// <returns>The property value interpreted as an integer.</returns>
     public int getPropertyAsInt(string key) => getPropertyAsIntWithDefault(key, 0);
 
     /// <summary>
     /// Get an Ice property as an integer.
-    /// If the property is not set, its default value is returned.
+    /// If the property is not set, its default value is returned. Throws PropertyException if the property is not a
+    /// known Ice property or the value is not a valid integer.
```
Comment (pepone) [inline csharp/src/Ice/Properties.cs:178]: ditto, move the exception docs to `exception` tag

**Review by pepone — APPROVED** [review]

**Pull request merged**